src/NUnitFramework/tests/Assertions/CollectionAssertTest.cs

```diff
@@ -47,14 +47,14 @@ public class CollectionAssertTest
         [Test()]
         public void ItemsOfType()
         {
-            var collection = new SimpleObjectCollection("x", "y", "z");
+            var collection = new SimpleEnumerable("x", "y", "z");
```

Comment:
CollectionAssert takes IEnumerable, so switch to use SimpleEnumerable to match.  Down below, I added IEquatable<T> to SimpleEnumerable<T> to match the correct behavior of Object.